core/src/main/java/org/apache/accumulo/core/client/summary/CountingSummarizer.java

```diff
@@ -0,0 +1,302 @@
+  public static final String MAX_CKL_DEFAULT = "128";
+
+  // this default can not be changed as persisted summary data depends on it
+  public static final String INGNORE_DELETES_DEFAULT = "true";
+
+  private static final String[] ALL_STATS = new String[] {TOO_LONG_STAT, TOO_MANY_STAT, EMITTED_STAT, SEEN_STAT, DELETES_IGNORED_STAT};
+
+  private int maxCounters;
+  private int maxCounterKeyLen;
+  private boolean ignoreDeletes;
+
+  private void init(SummarizerConfiguration conf) {
+    maxCounters = Integer.parseInt(conf.getOptions().getOrDefault(MAX_COUNTERS_OPT, MAX_COUNTER_DEFAULT));
+    maxCounterKeyLen = Integer.parseInt(conf.getOptions().getOrDefault(MAX_COUNTER_LEN_OPT, MAX_CKL_DEFAULT));
+    ignoreDeletes = Boolean.parseBoolean(conf.getOptions().getOrDefault(INGNORE_DELETES_OPT, INGNORE_DELETES_DEFAULT));
+  }
+
+  /**
+   * A function that converts key values to zero or more counter objects.
+   *
+   * @since 2.0.0
+   */
+  public static interface Converter<K> {
+    /**
+     * @param consumer
+     *          emit counter objects derived from key and value to this consumer
+     */
+    public void convert(Key k, Value v, Consumer<K> consumer);
+  }
+
+  /**
+   *
+   * @return A function that is used to convert each key value to zero or more counter objects. Each function returned should be independent.
+   */
+  protected abstract Converter<K> converter();
+
+  /**
+   * @return A function that is used to convert counter objects to String. The default function calls {@link Object#toString()} on the counter object.
+   */
+  protected Function<K,String> encoder() {
+    return Object::toString;
+  }
+
+  /**
+   * Override this if your key type is mutable and subject to change.
+   *
+   * @return a function that used to copy the counter object. This function is only used when the collector has never seen the counter object before. In this
+   *         case the collector needs to possibly copy the counter object before using as map key. The default implementation is the
+   *         {@link UnaryOperator#identity()} function.
+   */
+  protected UnaryOperator<K> copier() {
+    return UnaryOperator.identity();
+  }
+
+  @Override
+  public Collector collector(SummarizerConfiguration sc) {
+    init(sc);
+    return new Collector() {
+
+      // Map used for computing summary incrementally uses ByteSequence for key which is more efficient than converting String for each Key. The
+      // conversion to String is deferred until the summary is requested.
+
+      private Map<K,MutableLong> counters = new HashMap<>();
+      private long tooMany = 0;
+      private long tooLong = 0;
+      private long seen = 0;
+      private long emitted = 0;
+      private long deleted = 0;
+      private Converter<K> converter = converter();
+      private Function<K,String> encoder = encoder();
+      private UnaryOperator<K> copier = copier();
+
+      private void incrementCounter(K counter) {
+        emitted++;
```

Comment:
Currently it would bork the compaction, so the counter issue would be moot.

I did think about making behavior for failures configurable.  If a summarizer fails, we could record that fact that it ran and failed in the rfile OR let it bork the compaction ... depending on config.